### PR TITLE
Add 7TV global emotes

### PR DESCRIPTION
**Pull request by EdVraz**
resolves #5886 

~~Does not include real-time updating of global emotes as I was unable to get any response when adding/removing emotes.~~ Figured out a way to do it.

### CODE REVIEW

**Comment by NBruderman** [comment]
Don't forget to add the 7TV global emotes to the list in "src/modules/emote_autocomplete/twitch/EmoteAutocomplete.jsx"

**Comment by EdVraz** [comment]
> Don't forget to add the 7TV global emotes to the list in "src/modules/emote_autocomplete/twitch/EmoteAutocomplete.jsx" 

Oh I completely forgot to make a new emote category, will do that tomorrow

File: .eslintrc [hunk 1]
```diff
@@ -16,7 +16,12 @@
   "extends": ["airbnb", "prettier"],
   "plugins": ["prettier"],
   "rules": {
-    "prettier/prettier": ["error"],
```
Comment (dclstn) [inline .eslintrc:19]: Can you explain this change?

File: src/modules/seventv/global-emotes.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            continue;
+          }
+
+          this.emotes.set(code, createEmote(id, code, animated, owner, category));
+        }
+      })
+      .then(() => watcher.emit('emotes.updated'));
+
+    eventSource = new ReconnectingEventSource(
+      `https://events.7tv.io/v3@emote_set.update<object_id=62cdd34e72a832540de95857>`
```
Comment (dclstn) [inline src/modules/seventv/global-emotes.js:68]: I don't think updating global emotes in real-time is necessary.

File: src/modules/seventv/global-emotes.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import AbstractEmotes from '../emotes/abstract-emotes.js';
+import {createEmote} from './utils.js';
+import {EmoteCategories, EmoteProviders, EmoteTypeFlags, SettingIds} from '../../constants.js';
+import {hasFlag} from '../../utils/flags.js';
+import formatMessage from '../../i18n/index.js';
+
+// 7TV global emotes can be found here: https://7tv.app/emote-sets/62cdd34e72a832540de95857
+// they seem to be clean.
+
+// TODO emotes category
```
Comment (dclstn) [inline src/modules/seventv/global-emotes.js:13]: Looks like this is done?

File: .eslintrc (hunk 1 above)
Comment (EdVraz) [reply .eslintrc:19]: grafik
without it I get every line marked because the line separator is set to CLRF.

File: src/modules/seventv/global-emotes.js (hunk 2 above)
Comment (EdVraz) [reply src/modules/seventv/global-emotes.js:68]: okay

File: src/modules/seventv/global-emotes.js (hunk 3 above)
Comment (EdVraz) [reply src/modules/seventv/global-emotes.js:13]: yea

Comment (dclstn) [reply src/modules/seventv/global-emotes.js]: I would remove the comments above, too. The fetch below shows where they can be found.

File: .eslintrc (hunk 1 above)
Comment (EdVraz) [reply .eslintrc:19]: fixed

File: src/modules/seventv/global-emotes.js (hunk 3 above)
Comment (EdVraz) [reply src/modules/seventv/global-emotes.js]: Done

File: .eslintrc [hunk 4]
```diff
@@ -16,7 +16,9 @@
...
+    "prettier/prettier": [
```
Comment (night) [inline .eslintrc:19]: can you revert these changes?

File: src/constants.js [hunk 5]
```diff
@@ -155,6 +155,7 @@ export const EmoteCategories = {
   YOUTUBE_GLOBAL: 'youtube-global',
   YOUTUBE_CHANNEL: (channelId) => `youtube-${channelId}`,
   SEVENTV_CHANNEL: 'seventv-channel',
+  SEVENTV_GLOBAL: 'seventv-global',
```
Comment (night) [inline src/constants.js:158]: please sort this above the channel constant for consistency with the others

File: src/modules/emotes/index.js [hunk 6]
```diff
@@ -20,6 +21,7 @@ class EmotesModule {
       frankerfacezGlobalEmotes,
       frankerfacezChannelEmotes,
       seventvChannelEmotes,
+      seventvGlobalEmotes,
```
Comment (night) [inline src/modules/emotes/index.js:24]: order of this list matters so global emotes of each provider needs to come before channel emotes of each provider so channel emotes take preference

File: src/modules/seventv/channel-emotes.js [hunk 7]
```diff
@@ -90,14 +60,14 @@ class SevenTVChannelEmotes extends AbstractEmotes {
             continue;
           }
 
-          this.emotes.set(code, createEmote(id, code, animated, owner));
+          this.emotes.set(code, createEmote(id, code, animated, owner, category));
         }
       })
       .then(() => watcher.emit('emotes.updated'));
-
```
Comment (night) [inline src/modules/seventv/channel-emotes.js:97]: please re-add this line.

File: src/modules/seventv/channel-emotes.js [hunk 8]
```diff
@@ -90,14 +60,14 @@ class SevenTVChannelEmotes extends AbstractEmotes {
...
     eventSource = new ReconnectingEventSource(
       `https://events.7tv.app/v1/channel-emotes?channel=${encodeURIComponent(currentChannel.name)}`
     );
+
```
Comment (night) [inline src/modules/seventv/channel-emotes.js:70]: please remove this line.

File: src/modules/seventv/global-emotes.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    watcher.on('channel.updated', () => this.updateGlobalEmotes());
+    settings.on(`changed.${SettingIds.EMOTES}`, () => this.updateGlobalEmotes());
+  }
+
+  get category() {
+    return category;
+  }
+
+  updateGlobalEmotes() {
+    if (eventSource != null) {
```
Comment (night) [inline src/modules/seventv/global-emotes.js:30]: there does not appear to be an eventsource in this file at all. could we delete this and its other references?

File: src/modules/seventv/global-emotes.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  displayName: formatMessage({defaultMessage: '7TV Global Emotes'}),
+};
+
+let eventSource;
+
+class SevenTVGlobalEmotes extends AbstractEmotes {
+  constructor() {
+    super();
+
+    watcher.on('channel.updated', () => this.updateGlobalEmotes());
```
Comment (night) [inline src/modules/seventv/global-emotes.js:21]: why do we update global emotes when a channel updates?

**Review by night — CHANGES_REQUESTED** [review]

File: src/modules/seventv/global-emotes.js (hunk 10 above)
Comment (EdVraz) [reply src/modules/seventv/global-emotes.js:21]: Accidentally deleted the wrong line

**Review by night — APPROVED** [review]
